salt/state.py

```diff
@@ -1074,11 +1076,67 @@ def verify_data(self, data):
                                    req,
                                    data[u'__id__'])
                         errors.append(err)
+        errors += self.run_guards(data, guard_type='state')
```

Comment:
Here is my main question, the guards appear to get run during the state verification. I agree that then need to be after highstate compilation and before the state call, but would they be more useful right before the actual state function call? If we did than then a guard could make last minute modifications to the input values for a state as well as determine if the state should run.